### ISSUE TITLE
Backup time drifting may prevent backup being made in allowed hours

### ISSUE THREAD

**Issue by mikeynp** [lavinir/hassio-onedrive-backup#45]
I have my backup interval set to 1 day (24 hours) and the allowed hours set to 01:00-01:59. Each time a backup is made the time seems to drift a little, sometimes by a few minutes. For example, the last few backups I have are timestamped:
```
2023-01-27 01:16:08
2023-01-28 01:17:03
2023-01-29 01:17:21
2023-01-30 01:17:49
2023-01-31 01:18:25
2023-02-01 01:21:49
```
This may be because at each checking interval (currently 5 minutes) the time of the latest backup is checked to see if its older than 24 hours (in my case) and, if so, a backup is taken. Because of this the latest backup could be (at worst) the time of the last backup + 5 minutes, depending on when the check is made?

My theory is that if this drift continues, then the time for the latest backup will eventually be outside of the 01:00-01:59 allowed hours window. I think this would then cause the backup not to be made, leading to an additional 24 hours before the backup is eventually made (as it will then be > 24 hours old at the next allowed hours). 

If so, then its not the worst issue in the world as it looks like just one backup will be missed (at the correct time) every once in a while, but might be worth fixing if possible :-)

I'm running v1.41 and my configuration looks like this, if it helps:
```
recovery_mode: false
local_backup_num_to_keep: 7
onedrive_backup_num_to_keep: 7
backup_interval_days: 1
backup_name: hass_backup
monitor_all_local_backups: false
hass_api_timeout_minutes: 30
notify_on_error: true
exclude_media_folder: false
exclude_ssl_folder: false
exclude_share_folder: false
exclude_local_addons_folder: false
sync_paths: []
file_sync_remove_deleted: false
backup_allowed_hours: "1"
```

**Comment by lavinir** [comment]
Interesting. Theoretically I suppose you can miss a window if a backup is performed very near the end of the allowed hours window and the backup interval isn't able to be crossed within the window next time. This soudl indeed mean you would be stale until the next day.

I'll think about how to reduce chances for this.. In the meantime having more than an hour for your window would also help :)

**Comment by lavinir** [comment]
Hi, 
So using the start of the allowed hours wouldn't really be a good solution since this will break the backup freshness logic (allowed hours can be all day as well).
I am thinking of another solution to this issue where the intervals would just occur at constant times throughout the day instead of the drifting based on how long a cycle takes. 
I will likely include this in the next major version.

**Comment by mikeynp** [comment]
That makes sense. So you'd timestamp the new backup to the same time that each constant check took place, that way it wouldn't drift. Look forward to trying it out! Thanks.

**Comment by lavinir** [comment]
Fixed in version 2.0
